orttraining/orttraining/core/agent/training_agent.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <thread>
+#include <future>
+
+#include "core/common/common.h"
+#include "core/common/logging/logging.h"
+#include "core/framework/framework_common.h"
+#include "core/session/IOBinding.h"
+#include "core/session/inference_session.h"
+#include "orttraining/training_ops/cpu/controlflow/ort_tasks.h"
+
+
+namespace onnxruntime {
+namespace training {
+class IOBinding;
+
+// TODO: Define how minimal builds will play out here!
```

Comment:
the entire training agent should excluded from minimal build.

Reply:
+1. We do have an `ENABLE_TRAINING_OPS` macro that adds a minimal amount of training support to the minimal build, but we should exclude the TrainingAgent from that.

Reply:
done